api/src/game_state_server.py

```diff
@@ -175,20 +182,25 @@ async def process_updates(
                     )
                 else:
                     if not self._is_valid_token(token):
+                        logger.warning(f'Invalid token data {token}')
                         yield Message(
                             {client_id},
                             MessageContents(
                                 'error', f'Received a bad token: {data}', request_id
                             ),
                         )
-                    if not self._is_valid_position(token, room_id):
+                    if self._is_valid_position(token, room_id):
```

Comment:
Good bug fix, should be different PR. Side note: same bug exists just above this.

Reply:
reverted for this pr